File: 0003-longest-substring-without-repeating-characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        a = set()
        l = 0
        maxi = 0
        for i in range(len(s)):
            while s[i] in a:
                a.remove(s[l])
                l += 1
            a.add(s[i])
            maxi = max(maxi , i - l + 1)
        return maxi

File: 0003-longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max_len = 0;
        Stack<Character> stack = new Stack<Character>();
		int i = 0, j = 0;
		while(j < s.length()) {
			char ch = s.charAt(j);
			if(!stack.contains(ch)) {
				stack.push(ch);
				j++;
				max_len = (max_len > stack.size()) ? max_len : stack.size();
			}
			else {
				stack.removeFirst();
				i++;
			}
			
		}
		
		return max_len;
    }
}
